pythia/tasks/vqa/visual_genome/dataset.py

```diff
@@ -0,0 +1,204 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+import json
+import torch
+
+from pythia.tasks.vqa.vqa2 import VQA2Dataset
+from pythia.tasks.scene_graph_database import SceneGraphDatabase
+from pythia.common.sample import Sample, SampleList
+
+_CONSTANTS = {
+    "image_id_key": "image_id"
+}
+
+
+class VisualGenomeDataset(VQA2Dataset):
+    def __init__(self, dataset_type, imdb_file_index, config, *args, **kwargs):
+        super().__init__(dataset_type, imdb_file_index, config, *args, **kwargs)
+        self._name = "visual_genome"
+
+        self._return_scene_graph = config.return_scene_graph
+        self._return_objects = config.return_objects
+        self._return_relationships = config.return_relationships
+        self.scene_graph_db = None
+
+        should_build_scene_graph_db = (
+            self._return_scene_graph or self._return_objects
+            or self._return_relationships
+        )
+
+        if should_build_scene_graph_db:
+            scene_graph_file = config.scene_graph_files[dataset_type][imdb_file_index]
+            scene_graph_file = self._get_absolute_path(scene_graph_file)
+            self.scene_graph_db = SceneGraphDatabase(scene_graph_file)
+
+    def load_item(self, idx):
+        sample_info = self.imdb[idx]
+        sample_info = self._preprocess_answer(sample_info)
+        current_sample = Sample()
+
+        text_processor_argument = {"text": sample_info["question"]}
+
+        processed_question = self.text_processor(text_processor_argument)
+
+        current_sample.text = processed_question["text"]
+        current_sample.question_id = torch.tensor(
+            sample_info["id"], dtype=torch.int
+        )
+
+        if isinstance(sample_info["image_id"], int):
+            current_sample.image_id = torch.tensor(
+                sample_info["image_id"], dtype=torch.int
+            )
+        else:
+            current_sample.image_id = sample_info["image_id"]
+
+        current_sample.text_len = torch.tensor(
+            len(processed_question["text"]), dtype=torch.int
+        )
+
+        if self._use_features is True:
+            features = self.features_db[idx]
+            current_sample.update(features)
+
+        # Add details for OCR like OCR bbox, vectors, tokens here
+        current_sample = self.add_ocr_details(sample_info, current_sample)
+        # Depending on whether we are using soft copy this can add
+        # dynamic answer space
+        current_sample = self.add_answer_info(sample_info, current_sample)
+        current_sample = self._load_scene_graph(idx, current_sample)
```

Comment:
Is this almost same as `vqa2`'s `load_item` barring these two  lines? If so can we reduce code duplication?        
```
current_sample = self.add_answer_info(sample_info, current_sample)
current_sample = self._load_scene_graph(idx, current_sample) 
```

Reply:
There are other things too. VQA2's load_item takes in `question_tokens` instead of `question`, we can change it to use `tokens` but it might decrease some loading time. Also, `_preprocess_answer` is added as VG requires removing "." from the answers. That's why I decided to create a new `load_item` instead of using the one in `VQA2`.